nixos/modules/services/networking/firewall.nix

```diff
@@ -324,8 +324,12 @@ in
 
       package = mkOption {
         type = types.package;
-        default = pkgs.iptables;
-        defaultText = "pkgs.iptables";
+        default = if config.networking.nftables.enable
```

Comment:
How about doing `networking.firewall.package = mkDefault pkgs.iptables-nftables-compat;` in `networking.nftables` module?

Reply:
That's much nicer! Good plan :)

Reply:
Well, with your changes `pkgs.iptables-nftables-compat` will be used for anyone who use `nat` module, this will break stuff for sure. It should be used if `networking.nftables.enable` is `true`.